source/common/router/config_impl.h

```diff
@@ -7,9 +7,21 @@
 #include "envoy/runtime/runtime.h"
 #include "envoy/upstream/cluster_manager.h"
 
+#include "common/common/trie.h"
+#include "common/common/utility.h"
 #include "common/router/config_utility.h"
 
 namespace Router {
+// This is used to split a domain suffix into a series of path components (tokens) for placing in a
+// trie. E.g. ".foo.com" becomes {"com", ".", "foo", "."}. This is necessary because domain name
+// components proceed from most to least significant as right to left.
+struct ReverseDomainTokenizer {
+  std::vector<std::string> tokenize(const std::string& str) const {
+    std::vector<std::string> tokens(StringUtil::tokenize(str, "-."));
+    std::reverse(tokens.begin(), tokens.end());
```

Comment:
Is my mind playing tricks on me, or are we reversing both in the tokenizer and then in find/emplace again, thereby not reversing?